### ISSUE TITLE
Tensorflow Lite download_dependencies.sh doesn't work

### ISSUE THREAD

**Issue by rdepew** [tensorflow/tensorflow#35747]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian 10 (buster)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: Not a mobile device, it's a FreeWave ZumIQ edge computer. CPU is the same CPU as in a BeagleBone Black.
- TensorFlow installed from (source or binary): source
- TensorFlow version: 2.0.0
- Python version: 3.7
- Installed using virtualenv? pip? conda?: See description
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): gcc (Debian 8.3.0-6) 8.3.0
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the problem**

I'm trying to build a custom TF Lite wheel for this hardware. As a shortcut, I tried using the Raspberry Pi wheel, but the processor is different and so I got an "Illegal instruction" error. Therefore, I have to build my own.

I did this successfully last summer, with r1.14. It's really easy to do: I run the **download_dependencies.sh** script, then I do a couple of minor edits to the **build_pip_package.sh** script and run that script. 

Now it doesn't work anymore. It's the **download_dependencies.sh** that is causing the problem.

By the way, my shell is **/bin/bash** and my **uname-a** output is

`Linux fwt1311tp-min-jail-agent 4.4.0-170-generic #199-Ubuntu SMP Thu Nov 14 01:45:04 UTC 2019 armv7l GNU/Linux`

**Provide the exact sequence of commands / steps that you executed before running into the problem**

```
$ ./tensorflow/lite/tools/make/download_dependencies.sh
downloading https://gitlab.com/libeigen/eigen/-/archive/e6fcee995b0083e5652c79957090684a47a727c3/eigen-e6fcee995b0083e5652c79957090684a47a727c3.tar.gz

gzip: stdin: unexpected end of file
/bin/tar: Child returned status 1
/bin/tar: Error is not recoverable: exiting now
```
 
So I edit **download_dependencies.sh**. Lines 63 and 65 are:

```
63   if [[ "${url}" == *gz ]]; then
64  ...
55   elif [[ "${url}" == *zip ]]; then
```

I end up wrapping the **star gz** and **star zip** in double quotes.

This time, I get this far:

```
$ ./tensorflow/lite/tools/make/download_dependencies.sh
downloading https://gitlab.com/libeigen/eigen/-/archive/e6fcee995b0083e5652c79957090684a47a727c3/eigen-e6fcee995b0083e5652c79957090684a47a727c3.tar.gz
downloading https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/gemmlowp/archive/12fed0cd7cfcd9e169bf1925bc3a7a58725fdcc3.zip
downloading https://github.com/google/googletest/archive/release-1.8.0.tar.gz
downloading https://github.com/abseil/abseil-cpp/archive/43ef2148c0936ebf7cb4be6b19927a9d9d145b8f.tar.gz
downloading https://github.com/intel/ARM_NEON_2_x86_SSE/archive/master.zip
downloading https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/farmhash/archive/816a4ae622e964763ca0862d9dbd19324a1eaf45.tar.gz
downloading https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/flatbuffers/archive/v1.11.0.tar.gz
downloading https://storage.googleapis.com/mirror.tensorflow.org/www.kurims.kyoto-u.ac.jp/~ooura/fft2d.tgz
/bin/sed: can't read tensorflow/lite/tools/make/downloads/eigen/Eigen/src/Core/arch/NEON/Complex.h: No such file or directory
```
And, in fact, the directory **./tensorflow/lite/tools/make/downloads/eigen** is empty. All of the **downloads** directories are empty. Nothing was downloaded.

I `git checkout r1.14` instead, to see if it makes any difference. It doesn't. I get the same failures.

**Any other info / logs**

Rather than repeat the source code, I will point you to **tensorflow/lite/tools/make/download_dependencies.sh**.

**Comment by rdepew** [comment]
More information: I tried it on a different Linux system and it worked. Back to my hardware. I took the **curl** statement in line 64, omitted the **-s** option, and ran it from the command line:
```
curl -L https://gitlab.com/libeigen/eigen/-/archive/e6fcee995b0083e5652c79957090684a47a727c3/eigen-e6fcee995b0083e5652c79957090684a47a727c3.tar.gz | tar -xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

gzip: stdin: unexpected end of file
/bin/tar: Child returned status 1
/bin/tar: Error is not recoverable: exiting now
```
So the "silent" operation was masking the SSL certificate problem. After reading the page at https://curl.haxx.se/docs/sslcerts.html, I added the **-k** option to the **curl** commands on lines 64 and 69. That's a bandaid. I'll get the SSL certificate later.

This time, the files downloaded successfully, but immediately after downloading I got this error:
```
/bin/sed: can't read tensorflow/lite/tools/make/downloads/eigen/Eigen/src/Core/arch/NEON/Complex.h: Permission denied
```
I'm not sure what it's complaining about. I'm logged in as **devuser**, and **devuser** owns the file in question, and the directory as well:
```
$ whoami
devuser
$ ll tensorflow/lite/tools/make/downloads/eigen/Eigen/src/Core/arch/NEON
total 60
-rw-r--r-- 1 devuser devuser 19035 Jan 10 18:59 Complex.h
-rw-r--r-- 1 devuser devuser  1298 Jan  9 00:04 MathFunctions.h
-rw-r--r-- 1 devuser devuser 31476 Jan  9 00:04 PacketMath.h
-rw-r--r-- 1 devuser devuser  1338 Jan  9 00:04 TypeCasting.h
devuser@fwt1311tp-min-jail-agent:~/workspace/tensorflow$ whoami
devuser
$ ll -d tensorflow/lite/tools/make/downloads/eigen/Eigen/src/Core/arch/NEON
drwxr-xr-x 2 devuser devuser 4096 Jan 10 18:59 tensorflow/lite/tools/make/downloads/eigen/Eigen/src/Core/arch/NEON
devuser@fwt1311tp-min-jail-agent:~/workspace/tensorflow$
```

**Comment by rdepew** [comment]
I think I fixed it -- but it's a patch to a patch, and nothing to be proud of.

While examining the **download_dependencies.sh** code, I noticed that `replace_by_sed()` is called three times at the end of the script.  According to the comments for `replace_by_sed()` (_cheers to the person who put those comments there!_), the `replace_by_sed()` calls are a patch to make **download_dependencies.sh** work with iOS arm64. Since I'm compiling for Linux and armv7l, I concluded that this patch does not apply to my project, and so I commented out the calls to `replace_by_sed()`. This time, **download_dependencies.sh** completed successfully.

I know that a patch to a patch is not sustainable in the long term, but it worked for me.

**Comment by yyoon** [comment]
Thanks for the detailed report and your self investigation. I see 3 separate issues in your report.

1. `*gz` and `*zip` not quoted - We could fix this issue in the script itself.
2. SSL certificate issue - Sounds like this is something you should fix on your machine with the correct certificates installed.
3. permission issue with sed - I'm not sure why this is happening, and this would be difficult to reproduce without the actual hardware / OS setup you're using.

So, in summary I think we could fix 1, but not 2 and 3. Will send a patch soon.

**Comment by yyoon** [comment]
Actually, I think 1 should be quoted like `*"gz"` and `*"zip"`, with the wildcard outside of the double quotes. Did you also do this? or did you quote the entire string with the `*` character?

In the latter case, it is expected to not work correctly, because the `*` character is interpreted as a literal character, and not a wildcard.

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by bayesian-mind** [comment]
I also had the same issue and I solved by replacing the following line - 
`EIGEN_URL="$(grep -o 'http.*bitbucket.org/eigen/eigen/get/.*tar\.gz' "${BZL_FILE_PATH}" | grep -v mirror.tensorflow | head -n1)"`
with 
`EIGEN_URL="$(grep -o 'https.*bitbucket.org/eigen/eigen/get/.*tar\.gz' "${BZL_FILE_PATH}" | grep -v mirror.bazel | head -n1)"`
